src/main/java/duke/other/Parser.java

```diff
@@ -0,0 +1,62 @@
+package duke.other;
+
+import duke.command.AddCommand;
+import duke.command.Command;
+import duke.command.DeleteCommand;
+import duke.command.ShowCommand;
+
+/**
+ * Represents a Parser that extracts out the command type and its details from a full String command input by the user.
+ */
+public class Parser {
+    public Parser() {
+    }
+
+    /**
+     * Returns a Command object corresponding to the command type extracted.
+     *
+     * @param fullCommand Full String input by the user
+     * @return Command object corresponding to the command type(i.e.AddCommand if command type:Todo, Event or Deadline)
+     * @throws DukeException If the command type is invalid (i.e. hi, why)
+     */
+    public static Command parse(String fullCommand) throws DukeException {
+        String[] replyArr = fullCommand.split(" ");
+        String instruction = replyArr[0];
+        switch (instruction) {
+        case "bye":
+            Command c = new AddCommand(instruction, stringifyDetails(replyArr));
+            c.isExit = true;
+            System.out.println("    Bye! See ya later, alligator!");
+            return c;
+        case "delete":
+            return new DeleteCommand(instruction, replyArr);
+        case "deadline":
+        case "todo":
+        case "event":
+            return new AddCommand(instruction, stringifyDetails(replyArr));
+        case "list":
```

Comment:
again I think it'll be real good to externalises all this constant into a file! Reason being say the command "todo" changes to "doto" then you've got to go to every single instance of todo and make the change! if these constants were externalised to one file, every time a command changes wording you immediately know where to go!